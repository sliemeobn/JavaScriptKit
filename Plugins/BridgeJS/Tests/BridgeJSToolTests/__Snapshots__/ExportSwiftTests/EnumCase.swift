// NOTICE: This is auto-generated code by BridgeJS from JavaScriptKit,
// DO NOT EDIT.
//
// To update this file, just rebuild your project or run
// `swift package bridge-js`.

@_spi(BridgeJS) import JavaScriptKit

@_expose(wasm, "bjs_setDirection")
@_cdecl("bjs_setDirection")
public func _bjs_setDirection(direction: Int32) -> Void {
    #if arch(wasm32)
    setDirection(_: Direction(rawValue: Int(direction))!)
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_getDirection")
@_cdecl("bjs_getDirection")
public func _bjs_getDirection() -> Int32 {
    #if arch(wasm32)
    let ret = getDirection()
    return Int32(ret.rawValue)
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_processDirection")
@_cdecl("bjs_processDirection")
public func _bjs_processDirection(input: Int32) -> Int32 {
    #if arch(wasm32)
    let ret = processDirection(_: Direction(rawValue: Int(input))!)
    return Int32(ret.rawValue)
    #else
    fatalError("Only available on WebAssembly")
    #endif
}