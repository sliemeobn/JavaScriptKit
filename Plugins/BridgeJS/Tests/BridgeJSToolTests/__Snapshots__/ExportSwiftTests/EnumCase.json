{
  "classes" : [

  ],
  "enums" : [
    {
      "cases" : [
        {
          "associatedValues" : [

          ],
          "name" : "north"
        },
        {
          "associatedValues" : [

          ],
          "name" : "south"
        },
        {
          "associatedValues" : [

          ],
          "name" : "east"
        },
        {
          "associatedValues" : [

          ],
          "name" : "west"
        }
      ],
      "name" : "Direction",
      "swiftCallName" : "Direction"
    },
    {
      "cases" : [
        {
          "associatedValues" : [

          ],
          "name" : "loading"
        },
        {
          "associatedValues" : [

          ],
          "name" : "success"
        },
        {
          "associatedValues" : [

          ],
          "name" : "error"
        }
      ],
      "name" : "Status",
      "swiftCallName" : "Status"
    }
  ],
  "functions" : [
    {
      "abiName" : "bjs_setDirection",
      "effects" : {
        "isAsync" : false,
        "isThrows" : false
      },
      "name" : "setDirection",
      "parameters" : [
        {
          "label" : "_",
          "name" : "direction",
          "type" : {
            "caseEnum" : {
              "_0" : "Direction"
            }
          }
        }
      ],
      "returnType" : {
        "void" : {

        }
      }
    },
    {
      "abiName" : "bjs_getDirection",
      "effects" : {
        "isAsync" : false,
        "isThrows" : false
      },
      "name" : "getDirection",
      "parameters" : [

      ],
      "returnType" : {
        "caseEnum" : {
          "_0" : "Direction"
        }
      }
    },
    {
      "abiName" : "bjs_processDirection",
      "effects" : {
        "isAsync" : false,
        "isThrows" : false
      },
      "name" : "processDirection",
      "parameters" : [
        {
          "label" : "_",
          "name" : "input",
          "type" : {
            "caseEnum" : {
              "_0" : "Direction"
            }
          }
        }
      ],
      "returnType" : {
        "caseEnum" : {
          "_0" : "Status"
        }
      }
    }
  ],
  "moduleName" : "TestModule"
}